jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamCommon.kt

```diff
@@ -33,6 +34,27 @@ class SamCommon {
         // Exclusive
         val expectedSamMaxVersion = SemVer("0.16.0", 0, 16, 0)
 
+        private val versionCache = object : FileInfoCache<SemVer>() {
+            override fun getFileInfo(path: String): SemVer {
+                val commandLine = getSamCommandLine(path)
+                    .withParameters("--info")
+
+                val process = CapturingProcessHandler(commandLine).runProcess()
+
+                if (process.exitCode != 0) {
+                    throw IllegalStateException(process.stderr)
+                } else {
+                    if (process.stdout.isEmpty()) {
+                        message("lambda.run_configuration.sam.empty_info")
```

Comment:
Throw exception of this? This line doesn't do anything.

Reply:
Ah ya, old code did the same which caused the message to be returned, we should use same message but throw it so it doesnt cache